packages/fuels-types/src/errors.rs

```diff
@@ -56,6 +56,8 @@ pub enum Error {
     ValidationError(#[from] CheckError),
     #[error("Revert transaction error: {}, receipts: {:?}", .0, .1)]
     RevertTransactionError(String, Vec<Receipt>),
+    #[error("Tried to forward assets to a contract method that is not payable.")]
+    InvalidCallParameters(),
```

Comment:
`InvalidCallParameters` would indicate a validation issue for the call parameters, ie that the provided call parameters don't conform to the spec. This is not the case here. I think a better name could be `ForwardAssetToNonPayable` or something.

Reply:
Or `AssetsForwardedToNonPayableMethod`. 

Since the asset amount forwarding happens in the `CallParameters` they are kinda invalid. But, like @iqdecay , I'd make it more specific until we have at least one more reason why `CallParameters` might fail.